frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -303,17 +326,45 @@ export default ({
         }
       })
     },
-    setInitMessages () {
-      if (this.isJoinedChatRoom(this.currentChatRoomId)) {
-        this.messages = cloneDeep(this.chatRoomLatestMessages)
+    getSimulatedState (initialize = true) {
+      return {
+        settings: cloneDeep(this.chatRoomSettings),
+        attributes: cloneDeep(this.chatRoomAttributes),
+        users: cloneDeep(this.chatRoomUsers),
+        messages: initialize ? [] : this.messages,
+        saveMessage: true
+      }
```

Comment:
Oh, it looks like my comment here about using `cloneDeep` on the entire state instead of individually didn't get incorporated?

Reply:
Nevermind - it turns out that it must be done this way because in a chatroom with many messages, when we previous the chatroom, the chatroom constructor event hasn't been processed, and so it hasn't set the `initialState` to fill in the settings/attributes/users etc. The getters on the other hand will set those to empty objects/arrays.